File: app/src/main/java/com/gdi/hahi/ui/adapter/NeihanAdapter.java
package com.gdi.hahi.ui.adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.WindowManager;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.request.RequestOptions;
import com.gdi.hahi.R;
import com.gdi.hahi.callback.OnItemClickLitener;
import com.gdi.hahi.mvp.model.bean.HahiBean;

import java.util.List;

import fm.jiecao.jcvideoplayer_lib.JCVideoPlayerStandard;

/**
 * Created by Administrator on 2017/8/2 0002.
 */

public class NeihanAdapter extends RecyclerView.Adapter<NeihanAdapter.ViewHolder> {

    List<HahiBean> mList;
    Context mContext;
    OnItemClickLitener mOnItemClickLitener;
    int WindowWidth;

    public NeihanAdapter(Context context, List<HahiBean> mList) {
        this.mContext = context;
        this.mList = mList;
        init();
    }

    private void init() {
        WindowManager wm = (WindowManager) mContext.getSystemService(Context.WINDOW_SERVICE);
        WindowWidth = wm.getDefaultDisplay().getWidth();
//        Resources resources = mContext.getResources();
//        DisplayMetrics dm = resources.getDisplayMetrics();
//        WindowWidth= dm.widthPixels;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_neihan, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public int getItemCount() {
        return (mList != null) ? mList.size() : 0;
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, final int position) {
        if (mList != null) {
            HahiBean bean = mList.get(position);
            Glide.with(mContext).load(bean.getAvatar_url()).apply(new RequestOptions().circleCrop()).into(holder.iv_useravatar);
            holder.tv_username.setText(bean.getUser_name());
            if (!TextUtils.isEmpty(bean.getContext())){
                holder.tv_context.setText(bean.getContext());
            }else{
                holder.tv_context.setVisibility(View.GONE);
            }

            switch (bean.getMedia_type()) {
                case 0:
                    holder.iv_img.setVisibility(View.GONE);
                    holder.rl_video.setVisibility(View.GONE);
                    break;
                case 1:
                    holder.iv_img.setLayoutParams(setHeight(holder.iv_img, bean.getWidth(), bean.getHight()));
                    holder.iv_img.setVisibility(View.VISIBLE);
                    holder.rl_video.setVisibility(View.GONE);
                    Glide.with(mContext).load(bean.getMeida_url()).apply(new RequestOptions().centerCrop()).into(holder.iv_img);
                    break;
                case 2:
                    holder.iv_img.setLayoutParams(setHeight(holder.iv_img, bean.getWidth(), bean.getHight()));
                    holder.iv_img.setVisibility(View.VISIBLE);
                    holder.rl_video.setVisibility(View.GONE);
                    Glide.with(mContext).load(bean.getMeida_url()).apply(new RequestOptions().centerCrop()).into(holder.iv_img);
                    break;
                case 3:
                    holder.rl_video.setLayoutParams(setHeight(holder.rl_video, bean.getWidth(), bean.getHight()));
                    holder.gsy_player.setLayoutParams(setHeight(holder.gsy_player, bean.getWidth(), bean.getHight()));
                    holder.iv_img.setVisibility(View.GONE);
                    holder.rl_video.setVisibility(View.VISIBLE);
//                    bean.setMeida_url("http://pull-flv-l1-hs.pstatp.com/hudong/stream-6463304935256951565.flv");
                    holder.gsy_player.setUp(bean.getMeida_url(), JCVideoPlayerStandard.SCREEN_LAYOUT_NORMAL);
                    holder.gsy_player.backButton.setVisibility(View.GONE);
                    Glide.with(mContext).load(bean.getThumbImage()).apply(new RequestOptions().centerCrop()).into(holder.gsy_player.thumbImageView);
                    break;
            }


            holder.itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    if (mOnItemClickLitener != null) {
                        mOnItemClickLitener.setOnItemClickLitener(view, position);
                    }
                }
            });
        }
    }


    private ViewGroup.LayoutParams setHeight(View view, int width, int height) {
        ViewGroup.LayoutParams layoutParams = view.getLayoutParams();
        int lheight = 0;
        if (width!=0&&height!=0){
            lheight = height * WindowWidth / width;
            if (lheight>WindowWidth){
                lheight = WindowWidth;
            }
            layoutParams.height = lheight;
        }
        return layoutParams;
    }


    public class ViewHolder extends RecyclerView.ViewHolder {

        ImageView iv_useravatar;
        TextView tv_username;
        TextView tv_context;
        ImageView iv_img;
        RelativeLayout rl_video;
        JCVideoPlayerStandard gsy_player;

        public ViewHolder(View itemView) {
            super(itemView);
            iv_useravatar = (ImageView) itemView.findViewById(R.id.iv_useravatar);
            tv_username = (TextView) itemView.findViewById(R.id.tv_username);
            tv_context = (TextView) itemView.findViewById(R.id.tv_context);
            iv_img = (ImageView) itemView.findViewById(R.id.iv_img);
            rl_video = (RelativeLayout) itemView.findViewById(R.id.rl_video);
            gsy_player = (JCVideoPlayerStandard) itemView.findViewById(R.id.gsy_player);
        }
    }

    public void setOnItemClickLitener(OnItemClickLitener mOnItemClickLitener) {
        this.mOnItemClickLitener = mOnItemClickLitener;
    }
}

File: app/src/main/java/com/gdi/hahi/ui/adapter/NeihanAdapter.kt
package com.gdi.hahi.ui.adapter

import android.content.Context
import android.graphics.drawable.Drawable
import android.support.v7.widget.RecyclerView
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.WindowManager
import android.widget.ImageView
import android.widget.RelativeLayout
import android.widget.TextView
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions
import com.gdi.hahi.R
import com.gdi.hahi.callback.OnItemClickLitener
import com.gdi.hahi.mvp.model.bean.NeiHanData
import com.gdi.hahi.ui.activity.PhotoViewActivity
import com.shuyu.gsyvideoplayer.utils.ListVideoUtil
import com.shuyu.gsyvideoplayer.video.StandardGSYVideoPlayer


/**
 * Created by Administrator on 2017/8/2 0002.
 */

class NeihanAdapter(internal var mContext: Context, internal var mList: List<NeiHanData>?) : RecyclerView.Adapter<NeihanAdapter.ViewHolder>() {
    private var mOnItemClickLitener: OnItemClickLitener? = null
    private var WindowWidth: Int = 0
    private var viewHolderList: MutableList<RecyclerView.ViewHolder>? = null

    private var viewHolderMap: MutableMap<Int, ViewHolder>? = null

    init {
        val wm = mContext.getSystemService(Context.WINDOW_SERVICE) as WindowManager
        WindowWidth = wm.defaultDisplay.width
        viewHolderList = ArrayList()
        viewHolderMap = HashMap()
    }


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_neihan, parent, false)
        return ViewHolder(view)
    }

    override fun getItemCount(): Int {
        return if (mList != null) mList!!.size else 0
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        if (mList != null) {
            val bean = mList!![position]
            Glide.with(mContext).load(bean.avatar_url).apply(RequestOptions().circleCrop()).into(holder.iv_useravatar)
            holder.tv_username.text = bean.user_name
            if (!TextUtils.isEmpty(bean.context)) {
                holder.tv_context.text = bean.context
            } else {
                holder.tv_context.visibility = View.GONE
            }

            when (bean.media_type) {
                0 -> {
                    holder.iv_img.visibility = View.GONE
                    holder.rl_video.visibility = View.GONE
                }
                1 -> {
                    holder.iv_img.layoutParams = setHeight(holder.iv_img, bean.width, bean.hight)
                    holder.iv_img.visibility = View.VISIBLE
                    holder.rl_video.visibility = View.GONE
                    Glide.with(mContext).load(bean.meida_url).apply(RequestOptions().centerCrop()).into(holder.iv_img)
                }
                2 -> {
                    holder.iv_img.layoutParams = setHeight(holder.iv_img, bean.width, bean.hight)
                    holder.iv_img.visibility = View.VISIBLE
                    holder.rl_video.visibility = View.GONE
                    Glide.with(mContext).load(bean.meida_url).apply(RequestOptions().centerCrop()).into(holder.iv_img)
                }
                3 -> {
                    holder.rl_video.layoutParams = setHeight(holder.rl_video, bean.width, bean.hight)
                    holder.iv_img.visibility = View.GONE
                    holder.rl_video.visibility = View.VISIBLE
                    holder.gsy_player.setUp(bean.meida_url, false, null)
                    holder.gsy_player.backButton.visibility = View.GONE

                    //添加视预览图
                    var iv_thumb = ImageView(mContext)
                    Glide.with(mContext).asDrawable().load(bean.thumbImage).apply(RequestOptions().centerCrop()).into(iv_thumb)
                    holder.gsy_player.setThumbImageView(iv_thumb)

                    //全屏
                    holder.gsy_player.fullscreenButton.setOnClickListener {
                        //第一个true是否需要隐藏actionbar，第二个true是否需要隐藏statusbar
                        holder.gsy_player.startWindowFullscreen(mContext, true, true);
                    }

                }
            }
            //查看大图
            holder.iv_img.setOnClickListener {
                val intent = PhotoViewActivity.newUrlIntent(mContext,bean.meida_url)
                mContext.startActivity(intent)
            }

            holder.itemView.setOnClickListener { view ->
                if (mOnItemClickLitener != null) {
                    mOnItemClickLitener!!.setOnItemClickLitener(view, position)
                }
            }

            viewHolderMap?.let { viewHolderMap!!.put(position, holder) }
        }
    }


    private fun setHeight(view: View, width: Int, height: Int): ViewGroup.LayoutParams {
        val layoutParams = view.layoutParams
        var lheight = 0
        if (width != 0 && height != 0) {
            lheight = height * WindowWidth / width
            if (lheight > WindowWidth) {
                lheight = WindowWidth
            }
            layoutParams.height = lheight
        }
        return layoutParams
    }

     fun getViewHolder(postion: Int): ViewHolder? {
        if (viewHolderMap!!.get(postion)!==null){
            return viewHolderMap!!.get(postion)
        }
        return null
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        var iv_useravatar: ImageView
        var tv_username: TextView
        var tv_context: TextView
        var iv_img: ImageView
        var rl_video: RelativeLayout
        var gsy_player: StandardGSYVideoPlayer

        init {
            iv_useravatar = itemView.findViewById(R.id.iv_useravatar) as ImageView
            tv_username = itemView.findViewById(R.id.tv_username) as TextView
            tv_context = itemView.findViewById(R.id.tv_context) as TextView
            iv_img = itemView.findViewById(R.id.iv_img) as ImageView
            rl_video = itemView.findViewById(R.id.rl_video) as RelativeLayout
            gsy_player = itemView.findViewById(R.id.gsy_player) as StandardGSYVideoPlayer
        }
    }

    fun setOnItemClickLitener(mOnItemClickLitener: OnItemClickLitener) {
        this.mOnItemClickLitener = mOnItemClickLitener
    }
}
